intake/source/discovery.py

```diff
@@ -139,14 +153,7 @@ def enable(self, name, driver=None):
             Dotted object name, as in ``'intake_xarray.xzarr.ZarrSource'``.
             If None, simply remove driver disable flag, if it is found
         """
-        config = self.conf
-        if "drivers" not in config:
-            config["drivers"] = {}
-        if driver:
-            config["drivers"][name] = driver
-        elif config["drivers"].get(name) is False:
-            del config["drivers"][name]
-        config.save()
```

Comment:
enable/disable previously saved the altered config, so that te driver in question remained in the new state for the next session. That means the only way to do this now is via the CLI commands (which edit the config directly). Should save() be at least an option? I don't really see a downside to saving every time: if an ephemeral driver is not found next time, you can't load it whether it's in the enabled list or not.

Reply:
That makes sense. Currently the config is the only record of disabled plugins, so if the user decides to disable a runtime plugin or a discovered plugin then the change is reflected in the config. I was trying to make it so disabling a runtime plugin didn't affect the next session. Would it make sense to only save() if the plugin came from the config, or is the existing behavior preferable?

Reply:
I think if a runtime plugin is registered and then explicitly disabled, it's OK to save that in the config - if the config is then loaded in a new session without the plugin, it won't show up any way. Re-registering it will call enable again automatically?